yeonjae/programmers-0/선분의길이.js

```diff
@@ -0,0 +1,18 @@
+function solution(lines) {
+  let Counter = {};
+  let answer = 0;
+
+  for (const line of lines) {
+    const [x, y] = line;
+    x < y ? checkRange(x, y) : checkRange(y, x);
+  }
+  return answer;
+
+  function checkRange(start, end) {
+    for (let i = start; i < end; i++) {
+      Counter[i] = (Counter[i] || 0) + 1;
+
+      if (Counter[i] === 2) answer++;
+    }
+  }
+}
```

Comment:
```suggestion
function solution(lines) {
  const counterObj = {};
  let answer = 0;

  for (const line of lines) {
    const [x, y] = line;
    x < y ? checkRange(x, y) : checkRange(y, x);
  }
  return answer;

  function checkRange(start, end) {
    for (let i = start; i < end; i++) {
      counterObj[i] = (counterObj[i] || 0) + 1;

      if (counterObj[i] === 2) answer++;
    }
  }
}
```
- 풀이 잘봤습니다 🙇‍♂️ 그런데 왜 count 변수의 첫 글자를 대문자로..!? 하셨나요!? 또한, let이 아닌 const로 선언하셔야 됩니다 !!!
- 첫 글자를 대문자로 쓰는 경우는 class또는 생성자 함수를 이용할 때 입니다!